### ISSUE TITLE
Cannot get working on new install

### ISSUE THREAD

**Issue by derekoharrow** [jc21/nginx-proxy-manager#766]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image? YES
- Are you sure you're not using someone else's docker image? YES
- If having problems with Lets Encrypt, have you made absolutely sure your site is accessible from outside of your network? N/A

**Describe the bug**
Just installed and trying to get this working, but it fails to start - see log below.

I've set this up using the following docker-compose (temp port settings to not interfere with current nginx setup):
```
  npm:
    image: 'jc21/nginx-proxy-manager:latest'
    container_name: npm
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
    ports:
      - '9480:80'
      - '9481:81'
      - '9443:443'
    environment:
      DB_MYSQL_HOST: "mariadb"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "root"
      DB_MYSQL_PASSWORD: "mypasswordhere"
      DB_MYSQL_NAME: "npm"
    volumes:
      - /data/appdata/docker/npm/data:/data
      - /data/appdata/docker/npm/letsencrypt:/etc/letsencrypt
      - /data/appdata/docker/npm/config.json:/app/config/production.json
```

My MariaDB database is working fine for other containers. This is setup as follows:
```
  mariadb:
    image: "linuxserver/mariadb"
    container_name: "mariadb"
    hostname: mariadb
    volumes:
      - /etc/hosts:/etc/hosts:ro
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      - /data/appdata/docker/mariadb:/config
    ports:
      - target: 3306
        published: 3306
        protocol: tcp
        mode: host
    environment:
      - MYSQL_ROOT_PASSWORD=mymariadbpasswordhere
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
```

MariaDB version: _Server version: 10.4.17-MariaDB-1:10.4.17+maria~bionic-log - mariadb.org binary distribution_

I created an empty database in MariaDB called npm with a collation of utf8_general_ci.

On start, it created a config.json as follows:
```
{
  "database": {
    "fromEnv": true,
    "engine": "mysql",
    "host": "mariadb",
    "port": "3306",
    "user": "root",
    "password": "mysqlrootpassword",
    "name": "npm"
  }
```

When the container starts it creates tables within this DB, but then starts to produce errors as below, before the container fails and exits.

```
[services.d] starting services
[services.d] done.
❯ Enabling IPV6 in hosts: /etc/nginx/conf.d
  ❯ /etc/nginx/conf.d/include/block-exploits.conf
  ❯ /etc/nginx/conf.d/include/ssl-ciphers.conf
  ❯ /etc/nginx/conf.d/include/force-ssl.conf
  ❯ /etc/nginx/conf.d/include/ip_ranges.conf
  ❯ /etc/nginx/conf.d/include/proxy.conf
  ❯ /etc/nginx/conf.d/include/assets.conf
  ❯ /etc/nginx/conf.d/include/letsencrypt-acme-challenge.conf
  ❯ /etc/nginx/conf.d/include/resolvers.conf
  ❯ /etc/nginx/conf.d/production.conf
  ❯ /etc/nginx/conf.d/default.conf
❯ Enabling IPV6 in hosts: /data/nginx
[12/9/2020] [2:43:07 PM] [Migrate  ] › ℹ  info      Current database version: none
[12/9/2020] [2:43:07 PM] [Migrate  ] › ℹ  info      [initial-schema] Migrating Up...
[12/9/2020] [2:43:07 PM] [Migrate  ] › ℹ  info      [initial-schema] auth Table created
[12/9/2020] [2:43:08 PM] [Migrate  ] › ℹ  info      [initial-schema] user Table created
[12/9/2020] [2:43:10 PM] [Migrate  ] › ℹ  info      [initial-schema] user_permission Table created
[12/9/2020] [2:43:11 PM] [Migrate  ] › ℹ  info      [initial-schema] proxy_host Table created
[12/9/2020] [2:43:12 PM] [Migrate  ] › ℹ  info      [initial-schema] redirection_host Table created
[12/9/2020] [2:43:14 PM] [Migrate  ] › ℹ  info      [initial-schema] dead_host Table created
[12/9/2020] [2:43:14 PM] [Migrate  ] › ℹ  info      [initial-schema] stream Table created
[12/9/2020] [2:43:15 PM] [Migrate  ] › ℹ  info      [initial-schema] access_list Table created
[12/9/2020] [2:43:16 PM] [Migrate  ] › ℹ  info      [initial-schema] certificate Table created
[12/9/2020] [2:43:17 PM] [Migrate  ] › ℹ  info      [initial-schema] access_list_auth Table created
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [initial-schema] audit_log Table created
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [websockets] Migrating Up...
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [websockets] proxy_host Table altered
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [forward_host] Migrating Up...
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [forward_host] proxy_host Table altered
[12/9/2020] [2:43:18 PM] [Migrate  ] › ℹ  info      [http2_support] Migrating Up...
[12/9/2020] [2:43:19 PM] [Migrate  ] › ℹ  info      [http2_support] proxy_host Table altered
[12/9/2020] [2:43:19 PM] [Migrate  ] › ℹ  info      [http2_support] redirection_host Table altered
[12/9/2020] [2:43:19 PM] [Migrate  ] › ℹ  info      [http2_support] dead_host Table altered
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [forward_scheme] Migrating Up...
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [forward_scheme] proxy_host Table altered
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [disabled] Migrating Up...
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [disabled] proxy_host Table altered
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [disabled] redirection_host Table altered
[12/9/2020] [2:43:20 PM] [Migrate  ] › ℹ  info      [disabled] dead_host Table altered
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [disabled] stream Table altered
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [custom_locations] Migrating Up...
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [custom_locations] proxy_host Table altered
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [hsts] Migrating Up...
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [hsts] proxy_host Table altered
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [hsts] redirection_host Table altered
s6-svscanctl: fatal: unable to control /var/run/s6/services: supervisor not listening
finish: applet not found
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [hsts] dead_host Table altered
[12/9/2020] [2:43:21 PM] [Migrate  ] › ℹ  info      [settings] Migrating Up...
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
❯ Enabling IPV6 in hosts: /etc/nginx/conf.d
  ❯ /etc/nginx/conf.d/include/block-exploits.conf
  ❯ /etc/nginx/conf.d/include/ssl-ciphers.conf
  ❯ /etc/nginx/conf.d/include/force-ssl.conf
  ❯ /etc/nginx/conf.d/include/ip_ranges.conf
  ❯ /etc/nginx/conf.d/include/proxy.conf
  ❯ /etc/nginx/conf.d/include/assets.conf
  ❯ /etc/nginx/conf.d/include/letsencrypt-acme-challenge.conf
  ❯ /etc/nginx/conf.d/include/resolvers.conf
  ❯ /etc/nginx/conf.d/production.conf
  ❯ /etc/nginx/conf.d/default.conf
❯ Enabling IPV6 in hosts: /data/nginx
[12/9/2020] [2:43:30 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:30 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:31 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:31 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:32 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:32 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:33 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:33 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:34 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:34 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:35 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:35 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:36 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:36 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:37 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:37 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:38 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:38 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:39 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:39 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:40 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:40 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:41 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:41 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:42 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:42 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:43 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:44 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:45 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:45 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:46 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:46 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:47 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:47 PM] [Global   ] › ✖  error     Migration table is already locked
[12/9/2020] [2:43:48 PM] [Migrate  ] › ℹ  info      Current database version: 20190218060101
Can't take lock to run migrations: Migration table is already locked
If you are sure migrations are not running you can release the lock manually by deleting all the rows = require(migrations lock table: migrations_lock
[12/9/2020] [2:43:48 PM] [Global   ] › ✖  error     Migration table is already locked
s6-svscanctl: fatal: unable to control /var/run/s6/services: supervisor not listening
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
finish: applet not found
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
```

**Comment by cayetano** [comment]
The bug is related with '_DB_MYSQL_HOST: "mariadb"_'

If you use "db" as image name for my mysql image then work, nginx-manager image ignore DB_MYSQL_HOST: "mariadb", use "db"

**Comment by derekoharrow** [comment]
Not quite sure I'm with you? My DB host, running in another docker container, is called mariadb and is accessible by the name of mariadb. It manages to access the DB to create the tables, but then has the issue above, so DB access definitely works initially...

**Comment by derekoharrow** [comment]
If it helps, I'd prefer to use UNIX sockets, rather than TCP/IP for accessing the DB - how should this be configured to do that?

**Comment by derekoharrow** [comment]
I managed to trick it into doing the migrations - just by deleting the various tables from the DB and eventually it decided it was ok to continue. 

I'd suggest that there is an issue with using MariaDB with a fresh install without an existing DB.

**Comment by julesrulez96** [comment]
How did you manage to solve the Problem? @derekoharrow

**Comment by HackerWilson** [comment]
i don't know if there is a related issue which has been resolved or not.
but i found maybe the below text is the reason:

```
[7/9/2024] [12:44:01 PM] [Global   ] › ✖  error     Some plugins failed to install. Please check the logs above
[7/9/2024] [12:44:02 PM] [Migrate  ] › ℹ  info      Current database version: 20211108145214
[7/9/2024] [12:44:03 PM] [Global   ] › ⬤  debug     CMD: [ -f '/etc/letsencrypt/credentials/credentials-1' ] || { mkdir -p /etc/letsencrypt/credentials 2> /dev/null; echo '# Cloudflare API token
dns_cloudflare_api_token = XXXXX' > '/etc/letsencrypt/credentials/credentials-1' && chmod 600 '/etc/letsencrypt/credentials/credentials-1'; }
[7/9/2024] [12:44:03 PM] [Certbot  ] › ▶  start     Installing cloudflare...
[7/9/2024] [12:44:03 PM] [Global   ] › ⬤  debug     CMD: . /opt/certbot/bin/activate && pip install --no-cache-dir cloudflare acme==$(certbot --version | grep -Eo '[0-9](\.[0-9]+)+') certbot-dns-cloudflare==$(certbot --version | grep -Eo '[0-9](\.[0-9]+)+')  && deactivate
[7/9/2024] [12:44:05 PM] [Certbot  ] › ✖  error     The 'certbot_dns_cloudflare._internal.dns_cloudflare' plugin errored while loading: No module named 'CloudFlare'. You may need to remove or update this plugin. The Certbot log will contain the full error details and this should be reported to the plugin developer.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/certbot-log-95cki0p0/log or re-run Certbot with -v for more details.
The 'certbot_dns_cloudflare._internal.dns_cloudflare' plugin errored while loading: No module named 'CloudFlare'. You may need to remove or update this plugin. The Certbot log will contain the full error details and this should be reported to the plugin developer.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/certbot-log-ks6ncorw/log or re-run Certbot with -v for more details.
ERROR: Could not find a version that satisfies the requirement acme== (from versions: 0.0.0.dev20151006, 0.0.0.dev20151008, 0.0.0.dev20151017, 0.0.0.dev20151020, 0.0.0.dev20151021, 0.0.0.dev20151024, 0.0.0.dev20151030, 0.0.0.dev20151104, 0.0.0.dev20151107, 0.0.0.dev20151108, 0.0.0.dev20151114, 0.0.0.dev20151123, 0.0.0.dev20151201, 0.1.0, 0.1.1, 0.2.0, 0.3.0, 0.4.0, 0.4.1, 0.4.2, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.9.0, 0.9.1, 0.9.2, 0.9.3, 0.10.0, 0.10.1, 0.10.2, 0.11.0, 0.11.1, 0.12.0, 0.13.0, 0.14.0, 0.14.1, 0.14.2, 0.15.0, 0.16.0, 0.17.0, 0.18.0, 0.18.1, 0.18.2, 0.19.0, 0.20.0, 0.21.0, 0.21.1, 0.22.0, 0.22.1, 0.22.2, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.26.0, 0.26.1, 0.27.0, 0.27.1, 0.28.0, 0.29.0, 0.29.1, 0.30.0, 0.30.1, 0.30.2, 0.31.0, 0.32.0, 0.33.0, 0.33.1, 0.34.0, 0.34.1, 0.34.2, 0.35.0, 0.35.1, 0.36.0, 0.37.0, 0.37.1, 0.37.2, 0.38.0, 0.39.0, 0.40.0, 0.40.1, 1.0.0, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 1.10.0, 1.10.1, 1.11.0, 1.12.0, 1.13.0, 1.14.0, 1.15.0, 1.16.0, 1.17.0, 1.18.0, 1.19.0, 1.20.0, 1.21.0, 1.22.0, 1.23.0, 1.24.0, 1.25.0, 1.26.0, 1.27.0, 1.28.0, 1.29.0, 1.30.0, 1.31.0, 1.32.0, 2.0.0, 2.1.0, 2.2.0, 2.3.0, 2.4.0, 2.5.0, 2.6.0, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.8.0, 2.9.0, 2.10.0, 2.11.0)
ERROR: No matching distribution found for acme==

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: pip install --upgrade pip

[7/9/2024] [12:44:05 PM] [Global   ] › ✖  error     Some plugins failed to install. Please check the logs above
[7/9/2024] [12:44:06 PM] [Migrate  ] › ℹ  info      Current database version: 20211108145214
[7/9/2024] [12:44:06 PM] [Setup    ] › ℹ  info      Logrotate Timer initialized
[7/9/2024] [12:44:06 PM] [Global   ] › ⬤  debug     CMD: logrotate /etc/logrotate.d/nginx-proxy-manager
[7/9/2024] [12:44:07 PM] [Setup    ] › ℹ  info      Logrotate completed.
[7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
[7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
[7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
[7/9/2024] [12:44:08 PM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
[7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Let's Encrypt Renewal Timer initialized
[7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Renewing SSL certs expiring within 30 days ...
[7/9/2024] [12:44:09 PM] [IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
[7/9/2024] [12:44:09 PM] [Global   ] › ℹ  info      Backend PID 163 listening on port 3000 ...
[7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Completed SSL cert renew process
[7/9/2024] [12:44:44 PM] [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/default_host/site.conf
[7/9/2024] [12:44:44 PM] [Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "code": "ENOENT",
  "syscall": "unlink",
  "path": "/data/nginx/default_host/site.conf"
}
[7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
[7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
[7/9/2024] [12:44:44 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -s reload
```

some plugins failed to install, while `Backend PID 163 listening on port 3000` must be ready, then we can visit the npm login site  as normal....

@jc21

**Comment by HackerWilson** [comment]
> i don't know if there is a related issue which has been resolved or not. but i found maybe the below text is the reason:
> 
> ```
> [7/9/2024] [12:44:01 PM] [Global   ] › ✖  error     Some plugins failed to install. Please check the logs above
> [7/9/2024] [12:44:02 PM] [Migrate  ] › ℹ  info      Current database version: 20211108145214
> [7/9/2024] [12:44:03 PM] [Global   ] › ⬤  debug     CMD: [ -f '/etc/letsencrypt/credentials/credentials-1' ] || { mkdir -p /etc/letsencrypt/credentials 2> /dev/null; echo '# Cloudflare API token
> dns_cloudflare_api_token = XXXXX' > '/etc/letsencrypt/credentials/credentials-1' && chmod 600 '/etc/letsencrypt/credentials/credentials-1'; }
> [7/9/2024] [12:44:03 PM] [Certbot  ] › ▶  start     Installing cloudflare...
> [7/9/2024] [12:44:03 PM] [Global   ] › ⬤  debug     CMD: . /opt/certbot/bin/activate && pip install --no-cache-dir cloudflare acme==$(certbot --version | grep -Eo '[0-9](\.[0-9]+)+') certbot-dns-cloudflare==$(certbot --version | grep -Eo '[0-9](\.[0-9]+)+')  && deactivate
> [7/9/2024] [12:44:05 PM] [Certbot  ] › ✖  error     The 'certbot_dns_cloudflare._internal.dns_cloudflare' plugin errored while loading: No module named 'CloudFlare'. You may need to remove or update this plugin. The Certbot log will contain the full error details and this should be reported to the plugin developer.
> Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/certbot-log-95cki0p0/log or re-run Certbot with -v for more details.
> The 'certbot_dns_cloudflare._internal.dns_cloudflare' plugin errored while loading: No module named 'CloudFlare'. You may need to remove or update this plugin. The Certbot log will contain the full error details and this should be reported to the plugin developer.
> Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/certbot-log-ks6ncorw/log or re-run Certbot with -v for more details.
> ERROR: Could not find a version that satisfies the requirement acme== (from versions: 0.0.0.dev20151006, 0.0.0.dev20151008, 0.0.0.dev20151017, 0.0.0.dev20151020, 0.0.0.dev20151021, 0.0.0.dev20151024, 0.0.0.dev20151030, 0.0.0.dev20151104, 0.0.0.dev20151107, 0.0.0.dev20151108, 0.0.0.dev20151114, 0.0.0.dev20151123, 0.0.0.dev20151201, 0.1.0, 0.1.1, 0.2.0, 0.3.0, 0.4.0, 0.4.1, 0.4.2, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.9.0, 0.9.1, 0.9.2, 0.9.3, 0.10.0, 0.10.1, 0.10.2, 0.11.0, 0.11.1, 0.12.0, 0.13.0, 0.14.0, 0.14.1, 0.14.2, 0.15.0, 0.16.0, 0.17.0, 0.18.0, 0.18.1, 0.18.2, 0.19.0, 0.20.0, 0.21.0, 0.21.1, 0.22.0, 0.22.1, 0.22.2, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.26.0, 0.26.1, 0.27.0, 0.27.1, 0.28.0, 0.29.0, 0.29.1, 0.30.0, 0.30.1, 0.30.2, 0.31.0, 0.32.0, 0.33.0, 0.33.1, 0.34.0, 0.34.1, 0.34.2, 0.35.0, 0.35.1, 0.36.0, 0.37.0, 0.37.1, 0.37.2, 0.38.0, 0.39.0, 0.40.0, 0.40.1, 1.0.0, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 1.10.0, 1.10.1, 1.11.0, 1.12.0, 1.13.0, 1.14.0, 1.15.0, 1.16.0, 1.17.0, 1.18.0, 1.19.0, 1.20.0, 1.21.0, 1.22.0, 1.23.0, 1.24.0, 1.25.0, 1.26.0, 1.27.0, 1.28.0, 1.29.0, 1.30.0, 1.31.0, 1.32.0, 2.0.0, 2.1.0, 2.2.0, 2.3.0, 2.4.0, 2.5.0, 2.6.0, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.8.0, 2.9.0, 2.10.0, 2.11.0)
> ERROR: No matching distribution found for acme==
> 
> [notice] A new release of pip is available: 24.0 -> 24.1.2
> [notice] To update, run: pip install --upgrade pip
> 
> [7/9/2024] [12:44:05 PM] [Global   ] › ✖  error     Some plugins failed to install. Please check the logs above
> [7/9/2024] [12:44:06 PM] [Migrate  ] › ℹ  info      Current database version: 20211108145214
> [7/9/2024] [12:44:06 PM] [Setup    ] › ℹ  info      Logrotate Timer initialized
> [7/9/2024] [12:44:06 PM] [Global   ] › ⬤  debug     CMD: logrotate /etc/logrotate.d/nginx-proxy-manager
> [7/9/2024] [12:44:07 PM] [Setup    ] › ℹ  info      Logrotate completed.
> [7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
> [7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
> [7/9/2024] [12:44:07 PM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
> [7/9/2024] [12:44:08 PM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
> [7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Let's Encrypt Renewal Timer initialized
> [7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Renewing SSL certs expiring within 30 days ...
> [7/9/2024] [12:44:09 PM] [IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
> [7/9/2024] [12:44:09 PM] [Global   ] › ℹ  info      Backend PID 163 listening on port 3000 ...
> [7/9/2024] [12:44:09 PM] [SSL      ] › ℹ  info      Completed SSL cert renew process
> [7/9/2024] [12:44:44 PM] [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/default_host/site.conf
> [7/9/2024] [12:44:44 PM] [Nginx    ] › ⬤  debug     Could not delete file: {
>   "errno": -2,
>   "code": "ENOENT",
>   "syscall": "unlink",
>   "path": "/data/nginx/default_host/site.conf"
> }
> [7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
> [7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
> [7/9/2024] [12:44:44 PM] [Nginx    ] › ℹ  info      Reloading Nginx
> [7/9/2024] [12:44:44 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -s reload
> ```
> 
> some plugins failed to install, while `Backend PID 163 listening on port 3000` must be ready, then we can visit the npm login site as normal....
> 
> @jc21

it looks like the new version solved the problem: #3831
